generic/tests/netperf.py

```diff
@@ -136,8 +144,24 @@ def mtu_set(mtu):
                         "The ethenet device does not support jumbo," "cancel test"
                     )
 
-    vm = env.get_vm(params["main_vm"])
-    vm.verify_alive()
+    pinned_node = 0
+    host_numa = utils_misc.NumaInfo()
+    node_list = host_numa.online_nodes_withcpumem
+    try:
+        for node in node_list:
+            node_mem_free = int(host_numa.read_from_node_meminfo(node, "MemFree"))
+            mem = int(params["mem"])
+            mem_kb = mem * 1024
+            if node_mem_free > mem_kb:
+                pinned_node = node
+        params["qemu_command_prefix"] = f"numactl -m {pinned_node}"
+        vm_name = params["main_vm"]
+        env_process.preprocess_vm(test, params, env, vm_name)
+        vm = env.get_vm(vm_name)
+        vm.verify_alive()
+    except virt_vm.VMCreateError:
+        test.error("Not enough memory on the nodes to create the VM")
```

Comment:
Better to check error message includes related info before test.error, to avoid other risks.

Reply:
If the VM creation failed because of other issues(assert, missing args, others), that should not be a memory problem, but here you assume it fails by insufficient memory for all exception.

Reply:
I get the idea, please tell me if now looks better, I don't have this kind of host for test it :sweat_smile:

Reply:
test.error/cancel depends on you, or find the message to match expected info first, if matched, cancel it, otherwise raise error.

Reply:
Code updated and tested the cancel case by hard coding the node:
```
 (1/3) Host_RHEL.m9.u5.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.netperf.with_jumbo.host_guest.q35: STARTED
 (1/3) Host_RHEL.m9.u5.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.netperf.with_jumbo.host_guest.q35: CANCEL: The node: 7 used for VM pinning is not valid (23.93 s)
```